From 3e899dcd7bced1de65b59ce3a1aa7d9b8c78bc55 Mon Sep 17 00:00:00 2001
From: Michel Thebeau <Michel.Thebeau@windriver.com>
Date: Mon, 6 Nov 2023 19:28:52 +0000
Subject: [PATCH] Add manager pause request to helm values.yaml

Add manager.pause to values.yaml to allow a developer to pause the
execution of vault-manager.  This is intended as a debugging option.

The vault-manager pauses on any of the exit_on_trap safe points.  A
value of '1' pauses before vault-manager executes any code (other than
variable initialization).


Signed-off-by: Michel Thebeau <Michel.Thebeau@windriver.com>
---
 values.yaml | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/values.yaml b/values.yaml
index e9547fd..fc22705 100644
--- a/values.yaml
+++ b/values.yaml
@@ -111,6 +111,13 @@ manager:
   log:
     defaultLogLevel: 2
 
+  # Debugging option to setup pause request for vault manager on startup
+  # A pause_on_trap file will be created with the content of this value
+  # Values may include a positive integer matching a call of
+  # exit_on_trap
+  #
+  # pause: 1
+
 injector:
   # True if you want to enable vault agent injection.
   # @default: global.enabled
-- 
2.34.1

